### ISSUE TITLE
Kubernetes ingress controller daemonset disappears

### ISSUE THREAD

**Issue by muemich** [containous/traefik#941]
When creating an ingress controller using Trafik as a LB the pods get created, but after a couple of seconds they disappear.
The logs can be found here: http://130.211.64.108:5601/goto/2485b0e5312aa07110ce7695ea78e067

**Comment by emilevauge** [comment]
I don't see any error log from traefik. But I see:
`Resources are within the expected limits.`
`The expected resources are {Limits:map[cpu:{i:{value:0 scale:0} d:{Dec:0xc8204698c0} s: Format:DecimalSI} memory:{i:{value:0 scale:0} d:{Dec:0xc820469d40} s: Format:BinarySI}] Requests:map[memory:{i:{value:0 scale:0} d:{Dec:0xc820469d40} s: Format:BinarySI} cpu:{i:{value:0 scale:0} d:{Dec:0xc8204698c0} s: Format:DecimalSI}]}`

**Comment by muemich** [comment]
Here's the export:
`22:47:36.000
time="2016-12-07T21:47:36Z" level=debug msg="Received event from kubernetes map[object:map[kind:Endpoints apiVersion:v1 metadata:map[creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512] subsets:[]] type:MODIFIED]"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:36Z" level=debug msg="Received event from kubernetes map[object:map[kind:Endpoints apiVersion:v1 metadata:map[creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512] subsets:[]] type:MODIFIED]" 
"   
 insertId: "tyzf05for6wzf"   
 log: "traefik-ingress-lb"   
}
22:47:36.000
time="2016-12-07T21:47:36Z" level=debug msg="Skipping event from kubernetes map[type:MODIFIED object:map[apiVersion:v1 metadata:map[selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512 creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system] subsets:[] kind:Endpoints]]"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:36Z" level=debug msg="Skipping event from kubernetes map[type:MODIFIED object:map[apiVersion:v1 metadata:map[selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512 creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system] subsets:[] kind:Endpoints]]" 
"   
 insertId: "tyzf05for6wzg"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
shutting down, got signal: Terminated
{
 metadata: {…}   
 textPayload: "shutting down, got signal: Terminated
"   
 insertId: "epmy8nfoaj3ea"   
 log: "POD"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="I have to go... terminated"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="I have to go... terminated" 
"   
 insertId: "tyzf05for6wzh"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="Stopping server"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="Stopping server" 
"   
 insertId: "tyzf05for6wzi"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=debug msg="Waiting 30 seconds before killing connections on entrypoint http..."
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=debug msg="Waiting 30 seconds before killing connections on entrypoint http..." 
"   
 insertId: "tyzf05for6wzj"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="Server stopped"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="Server stopped" 
"   
 insertId: "tyzf05for6wzk"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="Shutting down"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="Shutting down" 
"   
 insertId: "tyzf05for6wzl"   
 log: "traefik-ingress-lb"   
}`

**Comment by muemich** [comment]
Here's the export:
22:47:36.000
time="2016-12-07T21:47:36Z" level=debug msg="Received event from kubernetes map[object:map[kind:Endpoints apiVersion:v1 metadata:map[creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512] subsets:[]] type:MODIFIED]"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:36Z" level=debug msg="Received event from kubernetes map[object:map[kind:Endpoints apiVersion:v1 metadata:map[creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512] subsets:[]] type:MODIFIED]" 
"   
 insertId: "tyzf05for6wzf"   
 log: "traefik-ingress-lb"   
}
22:47:36.000
time="2016-12-07T21:47:36Z" level=debug msg="Skipping event from kubernetes map[type:MODIFIED object:map[apiVersion:v1 metadata:map[selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512 creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system] subsets:[] kind:Endpoints]]"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:36Z" level=debug msg="Skipping event from kubernetes map[type:MODIFIED object:map[apiVersion:v1 metadata:map[selfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler uid:af215c73-bcc1-11e6-b9d7-42010af00144 resourceVersion:4512 creationTimestamp:2016-12-07T21:11:13Z annotations:map[control-plane.alpha.kubernetes.io/leader:{\"holderIdentity\":\"gke-486949053560-5d1898d9aa2046ca6bbb\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-12-07T21:11:13Z\",\"renewTime\":\"2016-12-07T21:47:36Z\",\"leaderTransitions\":0}] name:kube-scheduler namespace:kube-system] subsets:[] kind:Endpoints]]" 
"   
 insertId: "tyzf05for6wzg"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
shutting down, got signal: Terminated
{
 metadata: {…}   
 textPayload: "shutting down, got signal: Terminated
"   
 insertId: "epmy8nfoaj3ea"   
 log: "POD"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="I have to go... terminated"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="I have to go... terminated" 
"   
 insertId: "tyzf05for6wzh"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="Stopping server"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="Stopping server" 
"   
 insertId: "tyzf05for6wzi"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=debug msg="Waiting 30 seconds before killing connections on entrypoint http..."
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=debug msg="Waiting 30 seconds before killing connections on entrypoint http..." 
"   
 insertId: "tyzf05for6wzj"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="Server stopped"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="Server stopped" 
"   
 insertId: "tyzf05for6wzk"   
 log: "traefik-ingress-lb"   
}
22:47:37.000
time="2016-12-07T21:47:37Z" level=info msg="Shutting down"
{
 metadata: {…}   
 textPayload: "time="2016-12-07T21:47:37Z" level=info msg="Shutting down" 
"   
 insertId: "tyzf05for6wzl"   
 log: "traefik-ingress-lb"   
}

**Comment by lmickh** [comment]
I'm seeing similar behavior with Kubernetes 1.5.3 and both current versions of Traefik (v1.1.2 and v1.2.0-rc2).  The container starts and within a few seconds it starts to terminate.  Can provide log examples if needed.

**Comment by timoreimann** [comment]
Does `kubectl describe <the terminated Traefik pod>` provide anything helpful?

**Comment by lmickh** [comment]
It does not because the pod records are removed when it is terminated.  I've gone through describe of the pod and daemonset while they were active, but didn't see anything noteworthy.

**Comment by timoreimann** [comment]
@lmickh do the logs of the pod give anything? (You might need the `--previous` flag to see the logs of the terminated pod.)

**Comment by lmickh** [comment]
@timoreimann Here is a gist of the logs from both the pod and the kube system stuff as well.

https://gist.github.com/lmickh/423b19543cb0a6f94e8e5d62d69ce075

**Comment by timoreimann** [comment]
@lmickh Traefik clearly receives a SIGTERM here:

`"time=\"2017-03-16T16:16:18Z\" level=info msg=\"I have to go... terminated\" \n"`

By any chance, do you possibly have a liveness probe (mis-)configured?

Pinging @dtomcej for further ideas.

**Comment by lmickh** [comment]
No liveness probes set.  I'll look into adding some later today to see if that matters.

**Comment by timoreimann** [comment]
Sure, give it try. It shouldn't make a difference though since the kubelet will use the container state by default, and we can see from the logs that the container is up and running.

FWIW, we're running Traefik as a DaemonSet too with no problems whatsoever. Haven't had a single restart in two weeks.

Are you possibly experiencing [pod evictions](https://kubernetes.io/docs/concepts/cluster-administration/out-of-resource/) due to resource contention? Have you set request and/or resource limits? Do you run Traefik in the kube-system namespace or a different one?

There's always a chance for running into k8s bugs too.

**Comment by lmickh** [comment]
It does run in a the kube-system namespace.  There are no request/limits defined nor quotas on the namespaces.  I've set aside an empty node just to run this deamonset so I would think resources are not an issue.  Have not been able to find any logs or indications of a pod eviction.

I'm trying to make sure to eliminate any of the kube related possibilities, but I've managed to do this with the nginx ingress controller without issue.  I chimed in on this since it seemed like muemich had a similar issue.
